route/src/route_generator_worker.cpp

```diff
@@ -438,5 +438,91 @@ namespace route {
         return true;
     }
 
+bool RouteGeneratorWorker::crosstrack_error_check(const geometry_msgs::PoseStampedConstPtr& msg, lanelet::ConstLanelet current, carma_wm::TrackPos llt_track)
+{
+  if (!world_model_->getMap()) 
+  {
+    ROS_ERROR_STREAM("WMListener received a route before a map was available. Dropping route message.");
+    return true;
+  }
+
+   
+
+  auto route = lanelet::ConstLanelets();
+  for(auto id : route_msg_.route_path_lanelet_ids)
+  {
+    auto ll = world_model_->getMap()->laneletLayer.get(id);
```

Comment:
Is this just adding all the lanelets in the map to the the route check here? Or is this variable not actually describing the route? Might be a defect if the former, a misleading name if the latter.

Reply:
Yes, this section is meant to get all of the lanelets in the map. These lanelets will later be filtered by id based on whether or not they are part of the route (lines 496 -505).